### PR TITLE
[CDAP-16656] Improve performance of schema editor

### PR DESCRIPTION
**Pull request by ajainarayanan**
- By default collapses nested schema to avoid rendering the entire tree
- Implements lazy loading for schema editor
- Implement lazy loading for sqlSelectorWidget when opened in joiner

JIRA: https://issues.cask.co/browse/CDAP-16656
Build: https://builds.cask.co/browse/CDAP-URUT235-1

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx [hunk 1]
```diff
@@ -171,6 +188,105 @@ class SchemaContainer extends React.Component<ISchemaContainerProps, ISchemaCont
...
+  private io = new IntersectionObserver(
+    (entries) => {
+      let lastVisibleElement = this.state.fieldWindowSize;
+      for (const entry of entries) {
+        let id = entry.target.getAttribute('id');
+        id = id.split('-').pop();
+        const idInt = parseInt(id, 10);
+        if (entry.isIntersecting) {
+          lastVisibleElement =
+            idInt + 50 > this.state.fieldWindowSize ? idInt + DEFAULT_FIELD_WINDOW_SIZE : idInt;
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx]: I assume the 50 here is coming from DEFAULT_FIELD_WINDOW_SIZE? If yes, can we use the variable here instead?

File: cdap-ui/app/directives/complex-schema/complex-schema.js [hunk 2]
```diff
@@ -25,6 +25,74 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout, Sc
...
+      return;
+    }
+    vm.io = new IntersectionObserver(
+      (entries) => {
+        let lastVisibleElement = vm.windowSize;
+        for (const entry of entries) {
+          const id = entry.target.getAttribute('lazyload-id');
+          const numID = parseInt(id, 10);
+          if (entry.isIntersecting) {
+            lastVisibleElement = numID + 100 > vm.windowSize ? numID + vm.DEFAULT_WINDOW_SIZE : numID;
```
Comment (yukiej) [inline cdap-ui/app/directives/complex-schema/complex-schema.js]: Where is 100 coming from here?

File: cdap-ui/app/directives/complex-schema/embedded-schema-selector/embedded-schema-selector.js [hunk 3]
```diff
@@ -24,13 +24,14 @@ angular.module(PKG.name+'.commons')
...
-      isDisabled: '='
+      isDisabled: '=',
+      initCollapse: '='
     },
     bindToController: true,
     controller: function (SchemaHelper) {
       var vm = this;
       vm.checkComplexType = SchemaHelper.checkComplexType;
-      vm.expanded = true;
+      vm.expanded = vm.initCollapse === true ? false : true;
```
Comment (yukiej) [inline cdap-ui/app/directives/complex-schema/embedded-schema-selector/embedded-schema-selector.js]: maybe say !vm.initCollapse instead for conciseness?

File: cdap-ui/app/directives/complex-schema/complex-schema.js (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/directives/complex-schema/complex-schema.js]: Have replaced it with valid const

**Review by yukiej — APPROVED** [review]
Small nit: Please update license headers. Otherwise LGTM!

**Comment by ajainarayanan** [comment]
Green build: https://builds.cask.co/browse/CDAP-URUT235-3

**Pull request merged**